cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -219,7 +221,8 @@ public void createOrUpdate(Reconciliation reconciliation, ConfigMap assemblyCm,
         try {
             topicOperator = TopicOperator.fromConfigMap(assemblyCm);
         } catch (Exception e) {
-            return Future.failedFuture(e);
```

Comment:
I think it's better to return `Future.failedFuture(e)` than to return `Future.failedFuture(e.getMessage());`, because the latter will lose the exception type info, the exception type will become the super-unhelpful `NoStackTraceThrowable`. Did you change it for a reason?

Reply:
Yes, if I keep there `Future.failedFuture(e)`, failedFuture is processed in calling method and that causes printing stack trace.

Reply:
Do you mean in AbstractAssemblyOperator.reconcileAll() or `ClusterOperator.eventReceived()`? We should probably stick with `failedFuture(e)` and put in the necessary logic in both those places to just print the message if the exception is a `InvalidConfigMapException`, or print the message with stacktrace otherwise. Iow, I think those two places are the right place to be dealing with this exception.